### ISSUE TITLE
Improve mobile menus (navigation)

### ISSUE THREAD

**Issue by michelesbw** [sbwtech/sbw#68]
Received many complaints about navigation: too complex, hard to find things, etc..
Main issues are:
1) people don't find sidebar...don't realize it is at the bottom of the content and actually (ie. campaigns) way too much scrolling is needed to reach it;
2) people don't understand/use filters (ie in activiy page below tabs)

After much thinking about it I would like to try to:
1) Put current "Settings" menu in main menu (see screenshot below);
2) not show "Filter"
3) Adjust Z-index for tabs
4) (this would help a lot imho) - Current "Settings" icon should be replaced by "context menu". For example if user selects Blogs in main menu then the upper right icon should be blog icon and clicking on it should open blog's sidebar from right to left, same effect as for main menu.

![mobile-menu](https://cloud.githubusercontent.com/assets/5203511/25818619/743dd62a-342b-11e7-99e6-5c6b6a399dc0.jpg)

What do you think guys?

**Comment by feroda** [comment]
I agree we should take this as an high priority issue. The content of this issue is too long and too critical. Maybe we should better take a breathe and list menu entries one by one and see whether they are all needed or not. 

Afterwards, given the goals we have, we could place them in the right position.
(...basically follow invites from the meeting with the web designer / ux )

I would love to make a proposal for this issue because with these statements I am not of too much help

**Comment by michelesbw** [comment]
I don't understand your first sentence.
What I think is: we do not need to remove stuff but explain what they are by clearly indicating it on top of each menu "Your content" - "Community content" and "Context menu". People are not complain because of unwanted things but because they don't understand what they are for (ie. see 2 items for "Blogs" and don't understand why...we know it's one for their blogs and another for all community blogs).

**Comment by michelesbw** [comment]
My idea here below:
![michele-layout-ideas2](https://user-images.githubusercontent.com/5203511/33785121-02678f0a-dc64-11e7-90dc-898966fee01a.png)

**Comment by juho-jaakkola** [comment]
Otherwise sounds good, but I'm afraid the hamburger menu will be too long in the mobile version, if both owner_block menu and site menu will be in it.

But let's try it out, and see how it works.

**Comment by michelesbw** [comment]
As for mobile please see my comment above:
"as for Smartphones (responsive)
clicking the hamburger menu, as first there should be desktop's left bar as described above ("Your items" (+ administration) + settings + logout links) expanded and following "Community" menu (to be expanded with a tap)

current "Settings" icon should be replaced by "context menu" (desktop's right sidebar as above described).
For both left and right mobile menus we should take into consideration something like this cool one http://mmenu.frebsite.nl/"

**Comment by juho-jaakkola** [comment]
Groups have their own owner_block menu, that is visible when looking at stuff in the group context. Would the leftbar menu still display *also* the owner_block menu of the *user* on those pages?

![image](https://user-images.githubusercontent.com/883920/33983720-75c0090a-e0b5-11e7-89ea-f7eb45e4da49.png)

This would mean that both leftbar and rightbar have almost identical menu. One for the user and one for the group.

When creating new content in the group context, would the user understand the content is being created specifically inside a group (instead of personal content) if the user's own owner_block menu is visible?

**Comment by michelesbw** [comment]
To me the one in the picture is not a owner block but a context menu (group's) which should be kept there in right column as ii clearly states contents' owner entity being that group ("Group"  files, "group" pages, "group" wall, etc.).
I hope the naming doesn't confuse things as for Elgg but for me and users "Owner menu" means averything related to the user, all the rest is context menu (Blog's, Group's, ....)

**Comment by juho-jaakkola** [comment]
Nope, that is in fact the same owner_block menu. The labels just change depending on who is the page owner (user/group).

So if the group's owner_block menu must stay, I must display the same menu also to the left column and somehow hack it to show the user-version of the menu simultaneously with the group-version. The "page owner entity" is a global property, so it might get a bit tricky to display both menus at the same time.

**Comment by michelesbw** [comment]
Hmmm so "owner" refers to page while it should be related to content owner (Group's content not page-called-group's content)...too bad. Let's see what can be done but surely current navigation is very difficult to understand for users and frankly confusing we must admit...which leads to a bad UX.

**Comment by juho-jaakkola** [comment]
Facebook in fact works the same way as Elgg. The sidebar menu shows items related to of the page owner, which is either a user or a group.

The only difference is that the menu is on the left side, while in Elgg it is in the right side:

|User menu|Group menu|
|----------|----------|
|![image](https://user-images.githubusercontent.com/883920/34521125-620e8382-f08c-11e7-89ee-e0c7221d758a.png)|![image](https://user-images.githubusercontent.com/883920/34521132-690c46d8-f08c-11e7-8aa0-3dd6f3f489d0.png)|

|User menu|Group menu|
|----------|----------|
|![image](https://user-images.githubusercontent.com/883920/34521271-3337167c-f08d-11e7-8835-6044b23e4209.png)|![image](https://user-images.githubusercontent.com/883920/34521273-368e3648-f08d-11e7-9dd1-3dc7769dbc55.png)
|
